### PR TITLE
Do not ignore DatabaseException for Snowflake

### LINKED ISSUE #3984: Liquibase docker image 4.20 ignores SQL errors on Snowflake
### Search first

- [X] I searched and no similar issues were found

### Description

[Here's the link](https://forum.liquibase.org/t/liquibase-seems-to-ignore-sql-errors-on-snowflake/7927) to a detailed error report and some helpful answers on the Liquibase forums.

In brief:
- Liquibase docker image 4.19.1 and above ignore SQL syntax errors in formatted SQL changesets on Snowflake, treat the changeset as applied although Snowflake Query History clearly shows the statement failed, and add the checksum to the DATABASECHANGELOG table.
- the issue **may** be the 3.13.27 jdbc driver. Liquibase 4.19.0 uses the 3.13.26 jdbc driver and has no issues.

### Steps To Reproduce

Use (for example) the following changeset:
`-- liquibase formatted sql

--changeset author:0
CREATE TABLE testtable (id THISDATATYPEDOESNOTEXIST);`

We ran it through a GitLab CICD pipeline using Liquibase docker image 4.20 and connecting to a Snowflake database.

Our .properties file has the following entry:
driver: net.snowflake.client.jdbc.SnowflakeDriver

### Expected/Desired Behavior

Liquibase should detect that the SQL statement failed on Snowflake. 
The user should see this information in the logs, and the checksum should not be written to the DATABASECHANGELOG.

### Liquibase Version

4.20

### Database Vendor & Version

Snowflake

### Liquibase Integration

GitLab CICD

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

unknown

### Additional Context

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

### PR DESCRIPTION
**Pull request by filipelautert**
## Impact

- [X] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

Since Liquibase 4.19.1 , any SQL error raised on Snowflake is being silently ignored . This PR fixes the issue.

Requires https://github.com/liquibase/liquibase-pro-tests/pull/734 to be merged together.

Fixes #3984

### CODE REVIEW

File: liquibase-integration-tests/src/test/java/liquibase/dbtest/AbstractIntegrationTest.java [hunk 1]
```diff
@@ -969,6 +971,21 @@ public void testInvalidIncludeDoesntBreakLiquibase() throws Exception {
         assertFalse(lockService.hasChangeLogLock());
     }
 
+    @Test
+    public void testInvalidSqlThrowsException() throws Exception {
```
Comment (StevenMassaro) [inline liquibase-integration-tests/src/test/java/liquibase/dbtest/AbstractIntegrationTest.java:975]: I am not familiar with this class. Does this run against snowflake?

Comment (filipelautert) [reply liquibase-integration-tests/src/test/java/liquibase/dbtest/AbstractIntegrationTest.java:975]: It runs to all configured databases on liquibase, but snowflake is not one of them. To test snowflake it needs to be on harness or liquibase-pro-tests.

Comment (filipelautert) [reply liquibase-integration-tests/src/test/java/liquibase/dbtest/AbstractIntegrationTest.java:975]: Nice. So I'll keep those tests (as they work for the other local databases) , but will log a debt to handle Snowflake .

**Review by StevenMassaro — APPROVED** [review]

**Review by suryaaki2 — APPROVED** [review]

**Review by XDelphiGrl — APPROVED** [review]
Liquibase was "eating" a Snowflake database error thrown by improper specification to create a column remark on a view. This fix re-throws the exception. 
* Functional and test harness executions passing.
* No additional testing required.

APPROVED

**Pull request merged**